### PR TITLE
[SPARK-10630][SQL] Add a createDataFrame API that takes in a java list

### PR DESCRIPTION
**Pull request by holdenk**
It would be nice to support creating a DataFrame directly from a Java List of Row.

### CODE REVIEW

**Comment by holdenk** [comment]
cc @mengxr who filed the Jira

File: sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala [hunk 1]
```diff
@@ -477,6 +477,20 @@ class SQLContext(@transient val sparkContext: SparkContext)
...
+   * Creates a [[DataFrame]] from an [[java.util.List]] containing [[Row]]s using the given schema.
+   * It is important to make sure that the structure of every [[Row]] of the provided List matches
+   * the provided schema. Otherwise, there will be runtime exception.
+   *
+   * @group dataframes
+   * @since 1.6.0
+   */
+  @DeveloperApi
+  def createDataFrame(rows: java.util.List[Row], schema: StructType): DataFrame = {
+    createDataFrame(sparkContext.parallelize(rows.asScala), schema)
```
Comment (mengxr) [inline sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala]: Should create a local DataFrame using `LocalRelation`. See https://github.com/holdenk/spark/blob/SPARK-10630-create-DataFrame-from-Java-List/sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala#L385.

**Comment by mengxr** [comment]
LGTM. Merged into master. @holdenk Could you check existing Java unit tests using DataFrames and simplify the code with the new method? Thanks!

**Comment by holdenk** [comment]
@mengxr Sounds like a great idea :)

**Pull request closed without merging**